pennylane/ops/qubit.py

```diff
@@ -907,6 +910,44 @@ class Hermitian(Observable):
     num_params = 1
     par_domain = "A"
     grad_method = "F"
+    _eigs = {}
+
+    @classmethod
+    def eigvals(cls, Hmat):
+        """Return the eigenvalues of the specified Hermitian observable while
+        also storing the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the tensor product observable
+        """
+        Hkey = tuple(Hmat.flatten().tolist())
+        if Hkey not in cls._eigs:
+            # store the eigenvalues and eigenvectors corresponding to H in a
+            # dictionary, so that they do not need to be calculated later
+            w, U = np.linalg.eigh(Hmat)
+            cls._eigs[Hkey] = {"eigval": w, "eigvec": U}
+
+        # retrieve eigenvalues
+        return cls._eigs[Hkey]["eigval"]
```

Comment:
Oh, interesting that this had to be a class method. I think it makes sense though - a static method wouldn't work here, right?

Reply:
Yep, need to have access to ``cls._eigs``